src/ArmTemplates/Commands/Executors/ExtractorExecutor.cs

```diff
@@ -380,6 +387,79 @@ public async Task<Template<GroupTemplateResources>> GenerateGroupsTemplateAsync(
             return apiTemplate;
         }
 
+        /// <summary>
+        /// Generates parameters template in the desired folder
+        /// </summary>
+        /// <param name="baseFilesGenerationDirectory">name of base folder where to save output files</param>
+        /// <returns>generated parameters template</returns>
+        public async Task<Template> GenerateParametersTemplateAsync(
+            List<string> apisToExtract,
+            LoggerTemplateResources loggerResources,
+            BackendTemplateResources backendResources,
+            NamedValuesResources namedValuesResources,
+            string baseFilesGenerationDirectory)
+        {
+            this.logger.LogInformation("Started generation of parameters template...");
+
+            var templateParameters = await this.parametersExtractor.CreateMasterTemplateParameterValues(
+                apisToExtract,
+                this.loggerExtractor.Cache,
+                loggerResources,
+                backendResources,
+                namedValuesResources,
+                this.extractorParameters);
+
+            if (!templateParameters.Parameters.IsNullOrEmpty())
+            {
+                await FileWriter.SaveAsJsonAsync(
+                    templateParameters,
+                    directory: baseFilesGenerationDirectory,
+                    fileName: this.extractorParameters.FileNames.Parameters);
+            }
+
+            this.logger.LogInformation("Finished generation of parameters template...");
+            return templateParameters;
+        }
+
+        public async Task<Template<MasterTemplateResources>> GenerateMasterTemplateAsync(
+           string baseFilesGenerationDirectory,
+           ApiTemplateResources apiTemplateResources = null,
+           PolicyTemplateResources policyTemplateResources = null,
+           ApiVersionSetTemplateResources apiVersionSetTemplateResources = null,
+           ProductTemplateResources productsTemplateResources = null,
+           ProductApiTemplateResources productApisTemplateResources = null,
+           TagApiTemplateResources apiTagsTemplateResources = null,
+           LoggerTemplateResources loggersTemplateResources = null,
+           BackendTemplateResources backendsTemplateResources = null,
+           AuthorizationServerTemplateResources authorizationServersTemplateResources = null,
+           NamedValuesResources namedValuesTemplateResources = null,
+           TagTemplateResources tagTemplateResources = null)
+        {
+            if (string.IsNullOrEmpty(this.extractorParameters.LinkedTemplatesBaseUrl))
```

Comment:
```suggestion
            if (string.IsNullOrWhitespace(this.extractorParameters.LinkedTemplatesBaseUrl))
```

Reply:
Once we already had a bug with that. Because of not well designed parameters before, they are string, do you remember that? And you can only determine that they are of some value, if they are not null or not empty.

Just to show that it will be a pain at some moment - [view this vs-code-extension code, that uses resource-kit](https://github.com/microsoft/vscode-apimanagement/blob/e246072a85271f3ee13d2e562b7e03392a0979ba/src/commands/extract.ts#L145-L148). This is basically how the resource kit is launched.

I think we can change it in future by finally making input parameters of correct type. Because by now it doesn't matter what you are passing here - it just has to be not empty string and that's it

Reply:
Ouch. Okay. TY for the explanation.